src/Generator/LambdaGenerator.cs

```diff
@@ -83,16 +84,39 @@ public Task<SyntaxList<MemberDeclarationSyntax>> GenerateAsync(TransformationCon
                                 where (member as MethodDeclarationSyntax)?.Identifier.ValueText == "Handle"
                                 select (MethodDeclarationSyntax)member).FirstOrDefault();
 
+            var constructorArgs = from constructor in declaration.Members.OfType<ConstructorDeclarationSyntax>()
+                                  from parameter in constructor.ParameterList.Parameters
+                                  select parameter;
+
             if (handleMember == null)
             {
                 var descriptor = new DiagnosticDescriptor("LJ0001", "Handle Method Not Implemented", "Implement the Handle method to provide Lambda Function Handler code.", "Lambdajection", DiagnosticSeverity.Error, true);
                 var diagnostic = Diagnostic.Create(descriptor, Location.Create(declaration.SyntaxTree, declaration.Span));
                 progress.Report(diagnostic);
 
-                var source = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
-                source.Cancel();
+                Cancel(cancellationToken);
+                throw new GenerationFailureException();
+            }
 
-                throw new Exception("Lambda must implement handle method");
+            if (!includeFactories && constructorArgs.Any())
+            {
+                foreach (var arg in constructorArgs)

```

Comment:
Add release note about compilation failure vs runtime failure for including IAwsFactories without first adding AWSSDK.SecuriityToken as a dependency.